src/components/fields/DateField.js

```diff
@@ -136,39 +136,61 @@ class DateField extends AskField {
+    const fieldIdentifier = `${component}--${fieldNumber}`
+
     return (
       <div style={ styles.base }>
         <div style={ styles.dateFields }>
-          <input
-            ref={el => this._month = el}
-            type="number"
-            step="1"
-            placeholder="MM"
-            min="1"
-            max="12"
-            value={ this.state.month }
-            onChange={ this.onDateChange }
-            style={ this.getDateInputStyles('month') } />
-          <input
-            min="1"
-            max={this.daysInMonth()}
-            ref={el => this._day = el}
-            type="number"
-            step="1"
-            placeholder="DD"
-            value={ this.state.day }
-            onChange={ this.onDateChange }
-            style={ this.getDateInputStyles('day') } />
-          <input
-            ref={el => this._year = el}
-            type="number"
-            step="1"
-            placeholder="YYYY"
-            value={ this.state.year }
-            onChange={ this.onDateChange }
-            style={ this.getDateInputStyles('year') } />
+          <fieldset id={fieldIdentifier}>
+            <label for={`${fieldIdentifier}__month`} style={styles.visuallyhidden}>Month</label>
+            <input
+              id={`${fieldIdentifier}__month`}
+              ref={el => this._month = el}
+              type="number"
+              step="1"
+              placeholder="MM"
+              min="1"
+              max="12"
+              value={ this.state.month }
+              onChange={ this.onDateChange }
+              style={ this.getDateInputStyles('month') }
+            />
+
+            <label for={`${fieldIdentifier}__day`} style={styles.visuallyhidden}>Day</label>
+            <input
+              id={`${fieldIdentifier}__day`}
+              min="1"
+              max={this.daysInMonth()}
+              ref={el => this._day = el}
+              type="number"
+              step="1"
+              placeholder="DD"
+              value={ this.state.day }
+              onChange={ this.onDateChange }
+              style={ this.getDateInputStyles('day') }
+            />
+
+            <label for={`${fieldIdentifier}__year`} style={styles.visuallyhidden}>Year</label>
+            <input
+              id={`${fieldIdentifier}__year`}
+              ref={el => this._year = el}
+              type="number"
+              step="1"
+              placeholder="YYYY"
+              value={ this.state.year }
+              onChange={ this.onDateChange }
+              style={ this.getDateInputStyles('year') }
+            />
+          </fieldset>
         </div>
-        <div style={ styles.calendarButton } ref={el => this._calendarGroup = el} data-wrap data-clickOpens="false">
+        <div
+          role="button"
+          tabIndex="0"
+          style={ styles.calendarButton }
+          ref={el => this._calendarGroup = el}
+          data-wrap data-clickOpens="false"
```

Comment:
It might be good to include a description of this button for screen readers and make it accessible on keydown:

`<span style={styles.visuallyHidden}>Calendar Widget</span>`

Trick is, the default calendar widget isn't screen reader accessible (it just reads off a long list of dates), and fixing that seems like a broader task. I wonder if we should just make it aria-hidden for now, since there's another more convenient way to enter dates with a keyboard.